frameworks/Java/wicket/src/main/java/hellowicket/fortune/FortunePage.java

```diff
@@ -16,48 +16,42 @@
 import org.apache.wicket.markup.html.list.ListView;
 
 /**
- * A page that loads all fortune cookies
+ * A page that loads all fortune cookies. This mimics the Servlet example
+ * exactly, without any further optimizations that one could do such as sorting
+ * in the database, or not sorting at all.
  */
-public class FortunePage extends WebPage
-{
-  public FortunePage() throws Exception
-  {
-    List<Fortune> fortunes = new ArrayList<>(10000);
-
-    Fortune newFortune = new Fortune(0, "Additional fortune added at request time.");
-    fortunes.add(newFortune);
-
-    DataSource dataSource = WicketApplication.get().getDataSource();
-    try (Connection connection = dataSource.getConnection())
-    {
-      try (PreparedStatement statement = connection.prepareStatement("SELECT * FROM Fortune",
-              ResultSet.TYPE_FORWARD_ONLY,
-              ResultSet.CONCUR_READ_ONLY))
-      {
-          try (ResultSet resultSet = statement.executeQuery())
-          {
-              while (resultSet.next())
-              {
-                  fortunes.add(new Fortune(
-                          resultSet.getInt("id"),
-                          resultSet.getString("message")));
-              }
-          }
-      }
-    }
-
-    Collections.sort(fortunes);
-
-    ListView<Fortune> listView = new ListView<Fortune>("fortunes", fortunes)
-    {
-      @Override
-      protected void populateItem(ListItem<Fortune> item)
-      {
-        Fortune fortune = item.getModelObject();
-        item.add(new Label("id", fortune.id));
-        item.add(new Label("message", fortune.message));
-      }
-    };
-    add(listView);
-  }
+public class FortunePage extends WebPage {
+	private static final long serialVersionUID = 1L;
+
+	public FortunePage() throws Exception {
+		List<Fortune> fortunes = new ArrayList<>(10000);
+
+		DataSource dataSource = WicketApplication.get().getDataSource();
+		try ( //
+				Connection connection = dataSource.getConnection();
```

Comment:
No: this is the try-with resources statement that ignores exceptions upon close, so all resources claimed in the try() block are auto-closed. See https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html